src/test/mir-opt/issue-62289.rs

```diff
@@ -0,0 +1,82 @@
+// check that we clear the "ADT master drop flag" even when there are
+// no fields to be dropped.
+
+#![feature(box_syntax)]
+
+fn test() -> Option<Box<u32>> {
+    Some(box (None?))
+}
+
+fn main() {
+    test();
+}
+
+// END RUST SOURCE
+// START rustc.test.ElaborateDrops.before.mir
+// fn test() -> std::option::Option<std::boxed::Box<u32>> {
+//     ...
+//     bb0: {
+//         ...
```

Comment:
Is there a reason for omitting lines in so many places?

Reply:
I left out all the `StorageLive(...)` and `StorageDead(...)` lines but I can add them back in if you'd prefer a more exact test.

Reply:
I don't think there's any reason to omit them, given that the test is probably very sensitive to MIR generation changes without them.